packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -987,11 +993,24 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
       selectionGeometry: _selectionDelegate.value,
       onCopy: () {
         _copy();
-        hideToolbar();
+
+        // in Android copy should clear the selection
+        switch (defaultTargetPlatform) {
+          case TargetPlatform.android:
+            _clearSelection();
+            break;
+          case TargetPlatform.iOS:
+            hideToolbar(false);
+            break;
+          case TargetPlatform.fuchsia:
+          case TargetPlatform.linux:
+          case TargetPlatform.macOS:
+          case TargetPlatform.windows:
+            break;
+        }
```

Comment:
All the non-iOS platforms also hide the toolbar, right?  I feel like this might break the behavior on those platforms, though I haven't actually tried it.

Reply:
thank you for your comment, I clarify the switch case, only hide the toolbar in iOS and hide the selection in Android,  but other desktop platform do nothing. 
```
        switch (defaultTargetPlatform) {
          case TargetPlatform.android:
            _clearSelection();
            break;
          case TargetPlatform.iOS:
            hideToolbar(false);
            break;
          case TargetPlatform.fuchsia:
          case TargetPlatform.linux:
          case TargetPlatform.macOS:
          case TargetPlatform.windows:
            break;
        }
```

Reply:
That looks right to me, thanks!

Reply:
Oh also, would you mind adding a test that covers this behavior for desktop platforms?  That way we can catch things like this in the future.

Reply:
there was not the desktop test cases before, and I find it's wired to call `hideToolbar();` in desktop platforms, below the test case fail. why handlers are not null in in desktop platforms?
```
    // In Android copy should clear the selection.
    switch(defaultTargetPlatform) {
      case TargetPlatform.android:
        expect(regionState.selectionOverlay, isNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
        break;
      case TargetPlatform.iOS:
        expect(regionState.selectionOverlay, isNotNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNotNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNotNull);
        break;
      case TargetPlatform.linux:
      case TargetPlatform.macOS:
      case TargetPlatform.windows:
      case TargetPlatform.fuchsia:
        expect(regionState.selectionOverlay, isNotNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
        break;
    }
```

Reply:
of course, I would like to add tests, but now I have no idea to test start/end handlers in desktop platforms, would you give some advice? 

I tried the below code, but it doesn't work.
```
     // In Android copy should clear the selection.
    switch(defaultTargetPlatform) {
      case TargetPlatform.android:
        expect(regionState.selectionOverlay, isNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
        break;
      case TargetPlatform.iOS:
        expect(regionState.selectionOverlay, isNotNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNotNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNotNull);
        break;
      case TargetPlatform.linux:
      case TargetPlatform.macOS:
      case TargetPlatform.windows:
      case TargetPlatform.fuchsia:
        expect(regionState.selectionOverlay, isNotNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
        break;
    }
```

Reply:
Don't worry about the handles for desktop, I think we just need to test the toolbar:

```dart
     // In Android copy should clear the selection.
    switch(defaultTargetPlatform) {
      case TargetPlatform.android:
        expect(regionState.selectionOverlay, isNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNull);
        break;
      case TargetPlatform.iOS:
        expect(regionState.selectionOverlay, isNotNull);
        expect(regionState.selectionOverlay?.startHandleLayerLink, isNotNull);
        expect(regionState.selectionOverlay?.endHandleLayerLink, isNotNull);
        break;
      case TargetPlatform.linux:
      case TargetPlatform.macOS:
      case TargetPlatform.windows:
      case TargetPlatform.fuchsia:
        expect(regionState.selectionOverlay, isNotNull);
        break;
    }
```

Reply:
done~